### PR TITLE
[4.3] phpinfo disabled check

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue #40125

### Summary of Changes
Removes any spaces from the disable_functions string in the php.ini before exploding into an array for checking

### Testing Instructions
in your php.ini file add the following line and restart apache

` disable_functions = show_source, system, shell_exec, passthru, exec, phpinfo, popen, proc_open
`
then go to the joomla system dashboard and open the System Information

### Actual result BEFORE applying this Pull Request
image

### Expected result AFTER applying this Pull Request

image

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by richard67** [comment]
> in your php.ini file add the following line and restart apache

@brianteeman Which line?

**Comment by brianteeman** [comment]
> > in your php.ini file add the following line and restart apache
> 
> @brianteeman Which line?

oops - updated the instructions

File: administrator/components/com_admin/src/Model/SysinfoModel.php [hunk 1]
```diff
@@ -336,7 +336,9 @@ public function &getInfo(): array
      */
     public function phpinfoEnabled(): bool
     {
-        return !\in_array('phpinfo', explode(',', ini_get('disable_functions')));
+        // remove any spaces from the ini value before exploding it
+        $disabled = str_replace(' ', '', ini_get('disable_functions'));
```
Comment (joomdonation) [inline administrator/components/com_admin/src/Model/SysinfoModel.php]: It solves the issue. Maybe you would want to change the variable name to make it more meaningful (for example `$disabledFunctions` ) ? Let me know if you want to do that and I will report my test result.

Comment (brianteeman) [reply administrator/components/com_admin/src/Model/SysinfoModel.php]: renamed

File: administrator/components/com_admin/src/Model/SysinfoModel.php [hunk 2]
```diff
@@ -336,7 +336,9 @@ public function &getInfo(): array
      */
     public function phpinfoEnabled(): bool
     {
-        return !\in_array('phpinfo', explode(',', ini_get('disable_functions')));
+        // remove any spaces from the ini value before exploding it
+        $disabledFunctions = str_replace(' ', '', ini_get('disable_functions'));
+        return !\in_array('phpinfo', explode(',', $disabled));
```
Comment (joomdonation) [inline administrator/components/com_admin/src/Model/SysinfoModel.php]: Please change it on this line, too.

**Comment by toivo** [comment]
I have tested this item :red_circle: unsuccessfully on bfd54c6f4521e8cf35c6de78bd00f12fb7fc6422<br /><br />```
Warning: Undefined variable $disabled in C:\www\joomla43\administrator\components\com_admin\src\Model\SysinfoModel.php on line 341
Deprecated: explode(): Passing null to parameter #2 ($string) of type string is deprecated in C:\www\joomla43\administrator\components\com_admin\src\Model\SysinfoModel.php on line 341
```

Line 341 should be
```
        return !\in_array('phpinfo', explode(',', $disabledFunctions));
```
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40130">issues.joomla.org/tracker/joomla-cms/40130</a>.</sub>

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on ceaf637df695dd22c15a1cea2dbd2d79e9373f35<br /><br />Tested successfully in Joomla 4.3.0-beta5-dev of 18 March,<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40130">issues.joomla.org/tracker/joomla-cms/40130</a>.</sub>

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on ceaf637df695dd22c15a1cea2dbd2d79e9373f35<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40130">issues.joomla.org/tracker/joomla-cms/40130</a>.</sub>

**Comment by joomdonation** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40130">issues.joomla.org/tracker/joomla-cms/40130</a>.</sub>

**Comment by obuisard** [comment]
Thank you Brian @brianteeman for the PR!

**Comment by brianteeman** [comment]
thx

**Pull request merged**